### DISCUSSION TITLE
Windows Server, Vulnerabilities not showing CVEs

### DISCUSSION THREAD

**Discussion by DerGast01** [wazuh/wazuh#17294] [Q&A] [answered]
Hi at all!
I'm using Wazuh (the latest version, on Ubuntu) for a Windows Environment.
I've activated the Vulnarabilities Scan. It works fine for some Clients and Servers, but not for all! :)
I reinstalled the Wazuh Agent (4.4.2.1), but there is no input, although there is a "Last Scan time" showing.
Informations:
Windows Server 2019, last patches not installed! On another Server 2019 with the same Update Status, there are some Vulnerabilities showing.
The ossec Conf. on the Agent is the same like on the other Agents.
In my opinion, it is not a configuration failure.
I deleted this agent via CLI and reconnected it, but with the same fault.

![grafik](https://github.com/wazuh/wazuh/assets/134588318/cf072821-21cd-4b03-9aaa-922dfa85a49f)

Any ideas?

Kind regards
Daniel

**Comment by mdiego92** [comment]
Hello @DerGast01,

Could you please share with me the output of the following command from the Wazuh Manager node to which these Windows agents are connected?

`cat /var/ossec/logs/ossec.log | grep -iE 'vulnerability-detector'`

Also, could you please confirm if you have these configurations?

**Wazuh Manager ossec.conf file between the <vulnerability-detector> tags:**

```
    <provider name="msu">
      <enabled>yes</enabled>
      <update_interval>1h</update_interval>
    </provider>
```

**Wazuh Agent ossec.conf file or group configuration (centralized configuration):**

```
  <wodle name="syscollector">
    <disabled>no</disabled>
    <interval>1h</interval>
    <scan_on_start>yes</scan_on_start>
    <hardware>yes</hardware>
    <os>yes</os>
    <network>yes</network>
    <packages>yes</packages>
    <hotfixes>yes</hotfixes>
    <ports all="no">yes</ports>
    <processes>yes</processes>

    <!-- Database synchronization settings -->
    <synchronization>
      <max_eps>10</max_eps>
    </synchronization>
  </wodle>
```

Looking forward to your response.

**Reply by DerGast01** [reply]
This is the grep output. For all Agents the same:
![grafik](https://github.com/wazuh/wazuh/assets/134588318/9f92c0b3-5f29-4b70-ab89-94c963f86ffe)

![grafik](https://github.com/wazuh/wazuh/assets/134588318/57de15bb-436f-4514-b45a-3ed9ee036d7b)

Between the Tags :
![grafik](https://github.com/wazuh/wazuh/assets/134588318/cf1058d8-3c4c-4906-b867-5106d38e5da6)


The centralized Configuration:
![grafik](https://github.com/wazuh/wazuh/assets/134588318/aa137092-026f-4cb3-9757-aaa0b4298806)

**Reply by mdiego92** [reply]
Hello @DerGast01,

Could you please try adding the `<hotfixes>yes</hotfixes>` to the agent's syscollector configuration in the ossec.conf file?

Please remember to restart the Wazuh Agent after that.

Looking forward to your response!

**Reply by DerGast01** [reply]
hi mdiego92,
Thanks a lot!
I added the line and restarted the wazuh-manager
![grafik](https://github.com/wazuh/wazuh/assets/134588318/76ca08f1-9e2a-431c-8bfd-e9b6c7d014fc)

What should happen then? :)

wazuh-manager restart
Server restart
Same result:
![grafik](https://github.com/wazuh/wazuh/assets/134588318/3b93cb0b-9f0d-43ab-bb74-157489311259)

**Reply by mdiego92** [reply]
Hello  @DerGast01,

What's the ID for the Windows Agent?

Could you please enable the debugging options for the agent and restart it?

I would like to check the generated logs (ossec.log) from the agent if it's possible.

In the `internal_options.conf` file set (near the end of it):

windows.debug=2
analysisd.debug=2
wazuh_modules.debug=2

Restart the Wazuh Agent service.

**Reminder**: please revert these changes to the internal_options.conf file after getting the logs, as it can consume a lot of resources and disk space.

Looking forward to your response!

**Reply by DerGast01** [reply]
Hey :)
In this example it is the ID 003
003 is a Windows Server 2016 Datacenter without all patches.
Another Windows Server 2016 Datacenter without all patches, shows a few CVEs
In my opinion it's local problem? 
In the past i changed the Wazuh-Server IP-Address, but after this, i reinstalled the Wazuh Agent (with the correct Server IP) on all windows machines.
Maybe a checksum error in server db or anything else?

Now:
I stopped the wazuh-service
I activated the debug level.
I renamed the old ossec.log to ossec_old.log
Restart wazuh-service
A new ossec.log was generated.
Waiting till in the GUI shows "last partial Scan".
Then i stopped the wazuh-service.

![grafik](https://github.com/wazuh/wazuh/assets/134588318/645c925c-d005-4820-814d-90d565029704)

[ossec.log](https://github.com/wazuh/wazuh/files/11621984/ossec.log)

Thank you very much! :)

**Reply by mdiego92** [reply]
Hello @DerGast01,

I managed to reproduce your case with the same OS and Manager, but the configuration worked for me.

Could you please execute the following command from the manager the agent is connected to? It would allow us to know if the package inventory of the agent is populating data.

`sqlite3 /var/ossec/queue/db/003.db "select * from sys_programs;"
`

In case the output is empty, are you using a cluster configuration? In that case, all the workers have to have the same Vulnerability-Detector configuration.

Now, from the Wazuh Manager end, could you please add `wazuh_modules.debug=2` to `/var/ossec/etc/local_internal_options.conf` and then restart it? I would like to check the logs from the Manager.

**Reminder: please revert these changes to the internal_options.conf file after getting the logs, as it can consume a lot of resources and disk space.**

Looking forward to your response!

**Reply by DerGast01** [reply]
Hey :)
Thank you for your help!
It is not a cluster environment.

Here is the ossec.log. I hope, this is the correct one?
I deleted the old logfile for a clean diagnose and let it run for roundabout 10 Minutes. 
[ossec.log](https://github.com/wazuh/wazuh/files/11628412/ossec.log)


And this is the output from the table. There are many Entries:
![grafik](https://github.com/wazuh/wazuh/assets/134588318/e583d164-d928-49d2-b2b5-ae2ef308ed1d)

**Reply by mdiego92** [reply]
Hello @DerGast01,

Thank you very much for the log files. I'll perform some internal tests and provide an update as soon as possible.

Best regards!

**Reply by mdiego92** [reply]
Hello @DerGast01,

I tried reproducing your issue and for me, it is working without errors.

Could you please confirm if you're using a VPN? That could be causing problems obtaining network information.
Regarding the following errors:

```
2023/06/01 08:42:43 wazuh-agent[18088] cryptography.c:530 at verify_hash_and_pe_signature(): INFO: Trust verification of a module failed by using the hash method. CryptCATAdminEnumCatalogFromHash failed with error 1168 for 'C:\Program Files (x86)\ossec-agent\dbsync.dll': Element nicht gefunden.
2023/06/01 08:42:43 wazuh-agent[18088] dll_load_notify.c:90 at dll_notification(): WARNING: The file 'C:\Program Files (x86)\ossec-agent\dbsync.dll' is not signed or its signature is invalid.
```

Could you please share with me the signature data of the `C:\Program Files (x86)\ossec-agent\dbsync.dll` file?
![image](https://github.com/wazuh/wazuh/assets/62477966/18d51eae-9b15-48cd-af4d-95b37f768945)

Thank you very much for your patience and sorry for the late response.

**Comment by njmulsqb** [comment]
I am also facing this issue with windows server 2019

**Reply by DerGast01** [reply]
We have many Server 2019 Datacenter with the latest and not actually Updates.
I don't know why it works on 5  Windows Server 2019 but not on the others.
The only difference is that I reinstalled the agent on some Servers because the Wazuh server address was changed.. 
Hence my idea that something is wrong in the database on the Wazuh Server.
Maybe this can be recreated?

**Reply by njmulsqb** [reply]
You have marked my comment as answer brother, this may indicate to the maintainers that issue has been resolved hindering further potential support from them.

**Reply by DerGast01** [reply]
Thanks! Clicking to fast :D

**Comment by Tostti** [comment]
Question also asked in https://github.com/wazuh/wazuh-kibana-app/discussions/5495, may have some information for your research @mdiego92

**Comment by DerGast01** [comment]
Hey Diego! I'm so sorry for my late response. I was on vacation.
You are right!
The computers that have these problems were previously connected to another wazuh server via vpn for testing.
After testing i reinstalled the agent with the correct parameters for on-prem Wazuh-Server in the same Subnet.

![grafik](https://github.com/wazuh/wazuh/assets/134588318/458f4df7-6cbd-47a8-8515-28f4ce093b16)

**Comment by DerGast01** [comment]
Hey at all :)
Any ideas? The problem is still there...

@mdiego92 
For your question: "Could you please confirm if you're using a VPN? That could be causing problems obtaining network information."
At my first trial i had installed Wazuh Server on my HomeOffice (Virtual Box) and connected the Business Environment Server via VPN. That works well!
After this, i installed the Wazuh Server in the same Business Network Environment.
De/Reinstall the Agent on the Servers. And then, i had no connection.

**Comment by DerGast01** [comment, marked as answer]
Hey!
After a few weeks of testing, i decided to reinstall the Wazuh Application on the same Ubuntu Server.
Same IP Adresses, Clients/Agents not reinstalled... and it works.
After this, the Vulnerabilities are shown. So, in my opinion it was a Error/Failure in Wazuh Database?
